cpp/src/strings/convert/convert_durations.cu

```diff
@@ -0,0 +1,662 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/types.hpp>
+#include <strings/utilities.cuh>
+#include "utilities.cuh"
+
+#include <map>
+#include <vector>
+
+namespace cudf {
+namespace strings {
+namespace detail {
+
+namespace {
+
```

Comment:
I feel like there will be a lot of machinery in this file duplicated in the libcu++ implementation of std::chrono.  Did you look into that?

Reply:
I maximized the usage of `simt::std::chrono` with commit 40309f07f63cf0867a791aef4489c5774c676ad9. 
lambda `modulo_time` couldn't be replaced.